### ISSUE TITLE
What sensors does this support?

### ISSUE THREAD

**Issue by tylermiranda** [natekspencer/hacs-pentair#2]
Looking through the readme, I dont see what sensors it will create.  I'm looking to be able to control the on/off state on my pump as well as changing speeds through intellisync.  The integration does login but I'm only seeing the last report entities.

![image](https://github.com/natekspencer/hacs-pentair/assets/254684/117d7582-eb14-46e9-9863-875f81a66ffe)

**Comment by natekspencer** [comment]
Unfortunately, not a lot at this moment. Especially for your devices. This is a very rough draft implementation. I only have a salt level sensor, so my view is extremely limited. Others, like yourself, can enable debug logs and post that data as well as screenshots, and perform various actions to see how things change so that new sensors can be added. For control, someone will need to do even more by trying to figure out the network traffic/commands that are called.

**Comment by tylermiranda** [comment]
Thanks for replying. That's kind of what I figured...I'm at that point right now where I think I'm going to ditch my Intellisync altogether and build something using an ESP32 module., Reverse engineering the pentair home app as you've done is admirable but I'm also hesitant to put any work into it myself afraid that it will break...I'm really hating how Pentair locks their API behind their Screenlogic product which is too much $$$ IMO.

**Comment by natekspencer** [comment]
Yeah, I wish they had API specs and such. I hate being locked out from my own device's data. Fortunately for my device, I only need to read data, so the chances of bricking anything are negligible. Best of luck!

**Comment by mediacutlet** [comment]
Hey @natekspencer thanks for getting this custom integration off the ground! I'd like to contribute if I can. I recently had a IntelliFlo/Pro3 VSF installed and thought I'd give your component a try. It doesn't seem to pick up any sensor data. You mentioned enabled debug log, changing some values, and posting some screenshots. Would you mind explaining a bit more about that? Where can I enable and read the debug log? Is that from within the Pentair Home app or Home Assistant? Thanks for your time!
![image](https://github.com/natekspencer/hacs-pentair/assets/26329687/24e7765e-6ea4-416f-9e3b-4fcd31431de6)

**Comment by natekspencer** [comment]
@mediacutlet if you click on Pentair Home from your screenshot, you should find "Enable debug logging". The other screenshots I'm referring to would be from the Pentair Home app itself to potentially see what data is surfaced as well as changing values from the Pentair Home app to see how that data is reflected in the underlying api.

**Comment by mediacutlet** [comment]
Thanks for the info @natekspencer -- got it. Attached are two debug logs and a screenshot of the app interface. 

During the time the 01-56.608Z log was running I stopped the daily schedule, started it, disabled the whole pump, re-enabled it.

The second, file 06-09.585Z I stopped the daily schedule and then started it again.

Let me know if you think I should perform any specific action and record the debug log for it. Thanks again for your help! I'd love to eventually fork this or contribute once I learn a bit more.

[home-assistant_pentair_cloud_2024-04-17T21-01-56.608Z.log](https://github.com/natekspencer/hacs-pentair/files/15017217/home-assistant_pentair_cloud_2024-04-17T21-01-56.608Z.log)
[home-assistant_pentair_cloud_2024-04-17T21-06-09.585Z.log](https://github.com/natekspencer/hacs-pentair/files/15017218/home-assistant_pentair_cloud_2024-04-17T21-06-09.585Z.log)
![IMG_1941](https://github.com/natekspencer/hacs-pentair/assets/26329687/b81b352b-bcdd-435a-9580-92a2b40fdae9)

**Comment by natekspencer** [comment]
@mediacutlet I've released a beta version that includes some sensors if you can try it out and validate what shows in your app. If you use HACS, it's as simple as going into HACS and finding the Pentair Home integration, selecting `Redownload`, then `Show beta versions` and selecting the appropriate version and downloading. Please let me know what you see!

**Comment by mediacutlet** [comment]
My dude! This is incredible. Thank you.

![image](https://github.com/natekspencer/hacs-pentair/assets/26329687/a7145659-7a93-472d-8bc7-97d6dd14b0dc)

**Comment by natekspencer** [comment]
> My dude! This is incredible. Thank you.

Awesome, please play around with it and see how it matches with what the app displays. Additionally, if there are more screenshots you can share from the app, I might be able to figure out how to integrate some of the other data points. Also, please note this queries the API every 30 seconds, so it will be slightly delayed.

**Comment by mediacutlet** [comment]
All the values look spot on. There's not much else of value to show in the app, these are the only datapoints aside from this error/alert that popped up today. I started the app while debug was running and after about 10 seconds the alert popped up. Not sure if that could be a sensor "alert" on/off or number of alerts? It's perfect for my needs right now but since you asked and I saw the alert, here is the log and screenshot!

[home-assistant_pentair_cloud_2024-04-18T20-16-07.415Z.log](https://github.com/natekspencer/hacs-pentair/files/15030171/home-assistant_pentair_cloud_2024-04-18T20-16-07.415Z.log)

![pentair-error](https://github.com/natekspencer/hacs-pentair/assets/26329687/ce61be4a-35c3-4505-a7e8-ff66c11d0280)

**Comment by natekspencer** [comment]
I don't see anything in the log for it except for a weak RSSI. So they are either basing that alert on the RSSI/signal strength or a separate API call that I don't know about. Note: Signal strength is available as a diagnostic sensor, though disabled by default. So if that is important to you, you can enable and track it as well.

I'll probably push this to a prod release later today.

**Comment by stephywephy88** [comment]
I’ve been dying for a way to pull in my pool stuff! Would you be willing to look at my abbreviated debug log and see if the sensor adds are easy in exchange for some coffee funds? This part of log shows that I turned on my pool lights, changed my pump to 600rpm (but it primed to 3450 first), and fiddled with my salt water generator between 0-5%. Then turned everything off (left salt at 5%)

[Pentair Home Debug Logs.txt](https://github.com/natekspencer/hacs-pentair/files/15136121/Pentair.Home.Debug.Logs.txt)

Can also attach app screenshots as needed.

**Comment by natekspencer** [comment]
> Can also attach app screenshots as needed.

Screenshots will be helpful to see if there is other data I can also coordinate with the data reported in the log. But with your insight in the previous message, it looks like I could create sensors for at least the pump and salt water generator. Do the pool lights have brightness/colors/etc?

**Comment by stephywephy88** [comment]
Awesome! Yeah the lights I figured would be tricky - no brightness, but 5 colors and then stock themes (“color shows”). Some SS:

![IMG_5580](https://github.com/natekspencer/hacs-pentair/assets/47954870/589370f8-db2d-46b6-b792-04d62b8a8a28)
![IMG_5543](https://github.com/natekspencer/hacs-pentair/assets/47954870/751fa798-bf7e-462e-9150-abadd526163d)
![IMG_5548](https://github.com/natekspencer/hacs-pentair/assets/47954870/0597e5ef-abef-4f81-be12-3527d0fce76f)
![IMG_5545](https://github.com/natekspencer/hacs-pentair/assets/47954870/15a37e33-b93d-498d-95f3-9cfb6ce4a267)
